src/raw.rs

```diff
@@ -507,6 +562,9 @@ where
                 }
 
                 // Drop the task reference.
+                // SAFETY: This pointer is definitely alive because we hold a reference to it.
+                // TODO: But who holds the reference to it? The queue in general? The fact that
+                // it's scheduled?
```

Comment:
The reference is held by the `Waker` that is registered into the executor. See [here](https://github.com/smol-rs/async-executor/blob/57ffea098119504283e320f1a636183a13d9a071/src/lib.rs#L278).

Reply:
Done.